fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -166,7 +279,48 @@ public void rollback() throws PersistentStorageException {
             // No changes to rollback
             return;
         }
-        // rollback changes
+
+        if (!state.equals(State.COMMIT_FAILED) && !state.equals(State.PREPARE_FAILRED)) {
```

Comment:
Seems like you could "rollback" or at least abort a session in the `COMMIT_NOT_STARTED` state, it would only need to close sessions rather than rollback. Or are you thinking that a user requested transaction cancel would go through a different method?

Reply:
No - I think that makse sense. I'll change it.